poetry/installation/chef.py

```diff
@@ -75,13 +77,25 @@ def get_cached_archive_for_link(self, link):  # type: (Link) -> Optional[Link]
         return min(candidates)[1]
 
     def get_cached_archives_for_link(self, link):  # type: (Link) -> List[Link]
+
         cache_dir = self.get_cache_directory_for_link(link)
 
         archive_types = ["whl", "tar.gz", "tar.bz2", "bz2", "zip"]
         links = []
         for archive_type in archive_types:
             for archive in cache_dir.glob("*.{}".format(archive_type)):
-                links.append(Link(archive.as_uri()))
+
```

Comment:
adding a test to https://github.com/python-poetry/poetry/blob/master/tests/installation/test_chef.py would be nice to cement the contract that a bad link hash is not accepted

Reply:
done